testng-eclipse-plugin/src/main/org/testng/eclipse/util/LaunchUtil.java

```diff
@@ -284,19 +284,30 @@ public static void launchMethodConfiguration(IJavaProject javaProject,
           IMethod iMethod,
           String runMode,
           RunInfo runInfo) {
+    launchMethodConfiguration(javaProject, iMethod.getDeclaringType(), iMethod, runMode, runInfo);
+  }
+  
+  public static void launchMethodConfiguration(IJavaProject javaProject,
+          IType iType,
+          IMethod iMethod,
+          String runMode,
+          RunInfo runInfo) {
 
-    Set<IMethod> methods = Sets.newHashSet(iMethod);
+    Set<TypeAndMethod> typeAndMethods = Sets.newHashSet(new TypeAndMethod(iType, iMethod));

```

Comment:
now that `TypeAndMethod` is stored in `Set`, you need to override `equals()` and `hashCode()`